site/src/pages/WorkspacePage/useResourcesNav.ts

```diff
@@ -18,38 +18,36 @@ export const useResourcesNav = (resources: WorkspaceResource[]) => {
 
   const isSelected = useCallback(
     (resource: WorkspaceResource) => {
-      return resourceOptionId(resource) === resourcesNav.value;
+      return resourceOptionValue(resource) === resourcesNav.value;
     },
     [resourcesNav.value],
   );
 
-  const selectedResource = resources.find(isSelected);
-  const onSelectedResourceChange = useEffectEvent(
-    (previousResource?: WorkspaceResource) => {
+  const onResourceChanges = useEffectEvent(
+    (resources?: WorkspaceResource[]) => {
+      const hasSelectedResource = resourcesNav.value !== "";
+      const hasResources = resources && resources.length > 0;
       const hasResourcesWithAgents =
-        resources.length > 0 &&
-        resources[0].agents &&
-        resources[0].agents.length > 0;
-      if (!previousResource && hasResourcesWithAgents) {
-        resourcesNav.set(resourceOptionId(resources[0]));
+        hasResources && resources[0].agents && resources[0].agents.length > 0;
+      if (!hasSelectedResource && hasResourcesWithAgents) {
+        resourcesNav.set(resourceOptionValue(resources[0]));
       }
     },
   );
   useEffect(() => {
-    onSelectedResourceChange(selectedResource);
-  }, [onSelectedResourceChange, selectedResource]);
+    onResourceChanges(resources);
+  }, [onResourceChanges, resources]);
```

Comment:
In practice, this might make the effect run a little more often, but I think the logic for the main function is set up good, so the change shouldn't really be noticeable